### PR TITLE
Move tests from Mocha to Jest

### PR DESCRIPTION
**Pull request by JoeAmedeo**
These changes will move the existing tests from Mocha to Jest as a test framework, as described in #59 

The changes that went into this PR include:
- changing the syntax of `./__tests__/test.js` to work with jest
- replacing mocha / chai dependencies with jest dependencies
- updated test run script to use jest
- creating a jest configuration in the package.json
- adding the jest plugin to .eslintrc.json

__Additional Note__: As mentioned in the issue, the tests are still hanging after completion, but all tests should be passing regardless.

### CODE REVIEW

**Comment by JoeAmedeo** [comment]
Two quick questions, these might be out of scope for this PR, but just adding them in case

1) Should I add additional linter rules from the jest plugin? available rules [here](https://www.npmjs.com/package/eslint-plugin-jest#rules)
2) Should I add test coverage output?

File: package.json [hunk 1]
```diff
@@ -20,11 +20,14 @@
     "parse-link-header": "^1.0.1"
   },
   "devDependencies": {
-    "chai": "^4.2.0",
-    "eslint": "^7.10.0",
-    "eslint-config-prettier": "^6.12.0",
-    "eslint-plugin-prettier": "^3.1.4",
-    "mocha": "^8.1.3",
-    "prettier": "^2.1.2"
+    "eslint": "^6.8.0",
```
Comment (danisyellis) [inline package.json:23]: Our dependencies have been updated. Could you update yours to match?

**Review by danisyellis — COMMENTED** [review]
This looks great! I'm happy to merge it except for one small thing - I merged in a PR that updated our dependencies. Could you update yours to match? Thank you!

**Comment by danisyellis** [comment]
And to answer your questions:

> 1. Should I add additional linter rules from the jest plugin? available rules [here](https://www.npmjs.com/package/eslint-plugin-jest#rules)
Right now I don't think there are any that we need. I'm new to Jest, so let me know if there are any additional linter rules you think are particularly useful and I'll think about adding them in the future :-)
> 2. Should I add test coverage output?
That would be great! Thank you.

File: package.json (hunk 1 above)
Comment (JoeAmedeo) [reply package.json]: Should be up to date now

**Comment by JoeAmedeo** [comment]
I added test coverage output to the <rootDir>/coverage directory, and added the test:coverage script to trigger a coverage report.

in terms of linter rules, I usually just go with the recommended and style configurations.

**Pull request merged**